src/test/scala/ing/wbaa/druid/DruidQuerySpec.scala

```diff
@@ -156,6 +310,63 @@ class DruidQuerySpec extends WordSpec with Matchers with ScalaFutures {
     }
   }
 
+  "TopNQuery (streaming)" should {
+    "successfully be interpreted by Druid" in {
+      val threshold = 5
+
+      val query = TopNQuery(
+        dimension = Dimension(
+          dimension = "countryName"
+        ),
+        threshold = threshold,
+        metric = "count",
+        aggregations = List(
+          CountAggregation(name = "count")
+        ),
+        intervals = List("2011-06-01/2017-06-01")
+      )
+
+      val request = query.streamAs[TopCountry].runWith(Sink.seq[TopCountry])
+
+      whenReady(request) { topN =>
+        topN.size shouldBe threshold
+
+        topN.head shouldBe TopCountry(count = 35445, countryName = None)
+        topN(1) shouldBe TopCountry(count = 528, countryName = Some("United States"))
+        topN(2) shouldBe TopCountry(count = 256, countryName = Some("Italy"))
+        topN(3) shouldBe TopCountry(count = 234, countryName = Some("United Kingdom"))
+        topN(4) shouldBe TopCountry(count = 205, countryName = Some("France"))
+      }
+    }
+
+    "also work with a filter" in {
```

Comment:
Great to see you have spent time on writing extra tests!
I'm just wondering though for this (and similar ones). Do they add to the testing of streaming? 
It's only output related, correct?

I'm just thinking out loud here that if the tests are not essential in testing a specific new thing, it might be better to remove them to limit amount of maintainable tests. I'd rather have extra tests that test something we weren't already tested.

What do you think? And maybe I'm looking over it too fast and might be missing the point!

Reply:
You are right, it is only output related and thus it can be removed.